pkg/apicast/apicast.go

```diff
@@ -0,0 +1,357 @@
+package apicast
+
+import (
+	"net/url"
+	"strconv"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type APIcast struct {
+	Namespace              string
+	DeploymentName         string
+	ServiceName            string
+	Replicas               int32
+	AppLabel               string
+	ServiceAccountName     string
+	Image                  string
+	ExposedHost            ExposedHost
+	OwnerReference         *metav1.OwnerReference
+	AdminPortalCredentials AdminPortalCredentials
+
+	DeploymentEnvironment          *string
```

Comment:
Here I've used string instead of the custom defined type in the apicast_types.go as I don't want to be dependent on "external" types.